### PR TITLE
Update to axum 0.2

### PR DESCRIPTION
**Pull request by davidpdrsn**
Hey 👋 

I'm the maintainer of axum. For the last few weeks I've been working axum 0.2 which cleans up and improves pretty much all parts of axum. There is a number of breaking changes as well. Everything is detailed in our [changelog](https://github.com/tokio-rs/axum/blob/main/CHANGELOG.md).

While axum 0.2 isn't shipped yet, all the API level changes are done. I expect to `cargo publish` next week. As part of that I'm going through crates on crates.io to make sure everything still works on 0.2.

This PR currently uses a git dependency but once 0.2 is shipped then I'll remove that and depend directly on 0.2 from crates.io. Hence the draft PR.

Thanks for using axum! Just let me know if you have questions 😊

### CODE REVIEW

File: src/dtnd/ws.rs [hunk 1]
```diff
@@ -183,202 +183,221 @@ impl WsAASession {
...
-                    "/unsubscribe" => {
-                        if v.len() == 2 {
-                            if let Ok(eid) = EndpointID::try_from(v[1]) {
-                                if let Some(endpoints) = &mut self.endpoints {
-                                    endpoints.remove(&eid);
-                                    if let Some(ep) = (*DTNCORE.lock()).get_endpoint_mut(&eid) {
-                                        ep.clear_delivery_addr();
-                                    }
-                                    debug!("unsubscribed endpoint: {}", eid);
+        match msg {
```
Comment (davidpdrsn) [inline src/dtnd/ws.rs:187]: `Message` is an enum now, so matching on it is quite a bit easier.

File: Cargo.toml [hunk 2]
```diff
@@ -43,7 +43,7 @@ humansize = "1.1.0"
 sled = "0.34.4"
 enum_dispatch = "0.3.7"
 derive_more = "0.99.14"
-axum = { version="0.1.2", features = ["ws"] }
+axum = { git="https://github.com/tokio-rs/axum", branch = "main", features = ["ws"] }
```
Comment (davidpdrsn) [inline Cargo.toml:46]: I'll update this once 0.2 is released.

**Comment by gh0st42** [comment]
Thank you not only for axum itself but this great community service and the PR!
Looking forward to the 0.2 release!

**Comment by davidpdrsn** [comment]
Axum 0.2 is out. This is ready for review 🚀

**Pull request merged**